src/libcrun/container.c

```diff
@@ -3706,6 +3707,131 @@ libcrun_container_update_from_values (libcrun_context_t *context, const char *id
   return ret;
 }
 
+void populate_array_field(const char*** field, const char* array[], size_t numElements){
+    *field = malloc(numElements * sizeof(char *));
+    size_t i;
+    for (i = 0; i < numElements; i++) {
+        (*field)[i] = strdup(array[i]);
+    }
+
+    (*field)[i] = NULL;  // Terminate the array with NULL
+}
+
+int libcrun_container_print_features(struct features_info_s *info)
+{
+    // Hardcoded feature information
+    info->ociVersionMin = "1.0.0";
+    info->ociVersionMax = "1.0.2-dev";
+
+    // Hardcoded hooks
+    const char *hooks[] = {
+        "prestart",
+        "createRuntime",
+        "createContainer",
+        "startContainer",
+        "poststart",
+        "poststop"
+    };
+
+    // Hardcode the values for namespaces
+    static const char *namespaces[] = {
+        "cgroup",
+        "ipc",
+        "mount",
+        "network",
+        "pid",
+        "user",
+        "uts"
+    };
+
+   // Hardcode the values for capabilities
+    static const char *capabilities[] = {
+        "CAP_CHOWN",
+        "CAP_DAC_OVERRIDE",
+        "CAP_DAC_READ_SEARCH",
+        "CAP_FOWNER",
+        "CAP_FSETID",
+        "CAP_KILL",
+        "CAP_SETGID",
+        "CAP_SETUID",
+        "CAP_SETPCAP",
+        "CAP_LINUX_IMMUTABLE",
+        "CAP_NET_BIND_SERVICE",
+        "CAP_NET_BROADCAST",
+        "CAP_NET_ADMIN",
+        "CAP_NET_RAW",
+        "CAP_IPC_LOCK",
+        "CAP_IPC_OWNER",
+        "CAP_SYS_MODULE",
+        "CAP_SYS_RAWIO",
+        "CAP_SYS_CHROOT",
+        "CAP_SYS_PTRACE",
+        "CAP_SYS_PACCT",
+        "CAP_SYS_ADMIN",
+        "CAP_SYS_BOOT",
+        "CAP_SYS_NICE",
+        "CAP_SYS_RESOURCE",
+        "CAP_SYS_TIME",
+        "CAP_SYS_TTY_CONFIG",
+        "CAP_MKNOD",
+        "CAP_LEASE",
+        "CAP_AUDIT_WRITE",
+        "CAP_AUDIT_CONTROL",
+        "CAP_SETFCAP",
+        "CAP_MAC_OVERRIDE",
+        "CAP_MAC_ADMIN",
+        "CAP_SYSLOG",
+        "CAP_WAKE_ALARM",
+        "CAP_BLOCK_SUSPEND",
+        "CAP_AUDIT_READ",
+        "CAP_PERFMON",
+        "CAP_BPF",
+        "CAP_CHECKPOINT_RESTORE"
+    };
```

Comment:
this cannot be hardcoded.  We cannot guarantee it is kept in sync and that we know what capabilities are know to libcap.

I'd suggest doing something like:

```
cap_value_t i;
for (i = 0; ; i++)
    {
      char *v = cap_to_name(i);
      if (strstr(v, "_") == 0)
        break;
      append(v);
    }
```

even better if you make the test more robust and validate the returned name is not a number (you can try converting it with `strtol`.

Then make sure it is NULL terminated.  The simplest is to repeat the loop twice, once to calculate the size, then allocate the array, then fill it with another loop